From 009f8034b2bfb6c6e5c462454306f99f77f4c310 Mon Sep 17 00:00:00 2001
From: Alfred Klomp <git@alfredklomp.com>
Date: Fri, 2 May 2014 12:31:36 +0200
Subject: [PATCH 06/26] Bugfix: calculate proper size for memset()

The proper idiom is:

  memset(ptr, val, sizeof(*ptr));

These files were using:

  memset(ptr, val, sizeof(ptr));
---
 alg.c    | 4 ++--
 motion.c | 2 +-
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/alg.c b/alg.c
index 93c260f..343ff47 100644
--- a/alg.c
+++ b/alg.c
@@ -534,7 +534,7 @@ static int alg_labeling(struct context *cnt)
     imgs->labels_above = 0;
 
     /* Init: 0 means no label set / not checked. */
-    memset(labels, 0, width * height * sizeof(labels));
+    memset(labels, 0, width * height * sizeof(*labels));
     pixelpos = 0;
 
     for (iy = 0; iy < height - 1; iy++) {
@@ -1363,6 +1363,6 @@ void alg_update_reference_frame(struct context *cnt, int action)
         /* Copy fresh image */
         memcpy(cnt->imgs.ref, cnt->imgs.image_virgin, cnt->imgs.size);
         /* Reset static objects */
-        memset(cnt->imgs.ref_dyn, 0, cnt->imgs.motionsize * sizeof(cnt->imgs.ref_dyn)); 
+        memset(cnt->imgs.ref_dyn, 0, cnt->imgs.motionsize * sizeof(*cnt->imgs.ref_dyn));
     }
 }
diff --git a/motion.c b/motion.c
index ceaa44d..b650510 100644
--- a/motion.c
+++ b/motion.c
@@ -917,7 +917,7 @@ static int motion_init(struct context *cnt)
     /* Always initialize smart_mask - someone could turn it on later... */
     memset(cnt->imgs.smartmask, 0, cnt->imgs.motionsize);
     memset(cnt->imgs.smartmask_final, 255, cnt->imgs.motionsize);
-    memset(cnt->imgs.smartmask_buffer, 0, cnt->imgs.motionsize*sizeof(cnt->imgs.smartmask_buffer));
+    memset(cnt->imgs.smartmask_buffer, 0, cnt->imgs.motionsize * sizeof(*cnt->imgs.smartmask_buffer));
 
     /* Set noise level */
     cnt->noise = cnt->conf.noise;
-- 
1.8.1.4

